From ebe4bb86650e89d8540e6ee054b4ba20f0e3daac Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:58 -0400
Subject: [iommu] No need to pass '0x' when '%pa' is used

Message-id: <20140810182058.24055.5709.stgit@gir.stowe>
Patchwork-id: 87586
O-Subject: [RHEL7.1 PATCH 15/19] iommu: No need to pass '0x' when '%pa' is used
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: abedb049c52ef77ce7b11b915a4e7e6abd3985cb

commit abedb049c52ef77ce7b11b915a4e7e6abd3985cb
Author: Fabio Estevam <fabio.estevam@freescale.com>
Date:   Thu Aug 22 10:25:42 2013 -0300

    iommu: No need to pass '0x' when '%pa' is used

    Commit 6197ca82 (iommu: Use %pa and %zx instead of casting) introduced the
    usage of '%pa', but still kept the '0x', which leads to printing '0x0x'.

    Remove the '0x' when '%pa' is used.

    Signed-off-by: Fabio Estevam <fabio.estevam@freescale.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index d8c53c7..e5555fc 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -817,17 +817,17 @@ int iommu_map(struct iommu_domain *domain, unsigned long iova,
   * size of the smallest page supported by the hardware
   */
  if (!IS_ALIGNED(iova | paddr | size, min_pagesz)) {
-  pr_err("unaligned: iova 0x%lx pa 0x%pa size 0x%zx min_pagesz 0x%x\n",
+  pr_err("unaligned: iova 0x%lx pa %pa size 0x%zx min_pagesz 0x%x\n",
          iova, &paddr, size, min_pagesz);
   return -EINVAL;
  }
 
- pr_debug("map: iova 0x%lx pa 0x%pa size 0x%zx\n", iova, &paddr, size);
+ pr_debug("map: iova 0x%lx pa %pa size 0x%zx\n", iova, &paddr, size);
 
  while (size) {
   size_t pgsize = iommu_pgsize(domain, iova | paddr, size);
 
-  pr_debug("mapping: iova 0x%lx pa 0x%pa pgsize 0x%zx\n",
+  pr_debug("mapping: iova 0x%lx pa %pa pgsize 0x%zx\n",
     iova, &paddr, pgsize);
 
   ret = domain->ops->map(domain, iova, paddr, pgsize, prot);
-- 
1.7.1